Assets/Scripts/Behavior/IAttackBehavior.cs

```diff
@@ -4,10 +4,12 @@
 //
 // The attack behavior determines how the the threat navigates towards the asset.
 public interface IAttackBehavior {
+  IAgent Agent { get; set; }
```

Comment:
Recommend removing the setter such that the Agent can only be initialized during construction. If Agent becomes null after construction NREs will throw all over the place.

Reply:
Done in #116